linkerd/tls/src/imp/openssl.rs

```diff
@@ -0,0 +1,295 @@
+            Ok(_) => Ok(s),
+            Err(err) => Err(io::Error::new(ErrorKind::Other, err)),
+        }
+    }
+    #[cfg(feature = "boring-tls")]
+    pub async fn accept<IO>(&self, stream: IO) -> Result<server::TlsStream<IO>>
+    where
+        IO: AsyncRead + AsyncWrite + Unpin,
+    {
+        match tokio_boring::accept(&self.0, stream).await {
+            Ok(ss) => Ok(ss.into()),
+            Err(err) => {
+                let _ = err.ssl();
+                Err(io::Error::new(ErrorKind::Other, "Ble"))
+            }
+        }
+    }
+}
+
+impl From<SslAcceptor> for TlsAcceptor {
+    fn from(acceptor: SslAcceptor) -> Self {
+        Self(acceptor)
+    }
+}
+
+impl From<Arc<ServerConfig>> for TlsAcceptor {
+    fn from(conf: Arc<ServerConfig>) -> Self {
+        println!("SSL Version {}", version::version());
+        match fips::enable(true) {
+            Err(err) => tracing::error!("FIPS mode can not be enabled {:?}", err),
+            _ => debug!("FIPS mode is enabled"),
+        }
+
+        println!("Setting up ssl acceptor with mozilla intermediate");
+        let mut builder = SslAcceptor::mozilla_intermediate(SslMethod::tls()).unwrap();
+
+        let key = conf.0.key.as_ref().unwrap().as_ref().0.as_ref();
+        println!("Setting private key {:?}", key);
+        builder.set_private_key(key).unwrap();
+
+        let cert = conf.0.cert.as_ref().unwrap().as_ref();
+        println!("Setting certificate {:?}", cert);
+        builder.set_certificate(cert.cert.as_ref()).unwrap();
+
+        for cert in &cert.chain {
+            println!("Adding extra chain certificate {:?}", cert);
+            builder.add_extra_chain_cert(cert.clone()).unwrap()
+        }
+
+        builder.set_cert_store(X509StoreBuilder::new().unwrap().build());
+        conf.0
+            .root_certs
+            .objects()
+            .iter()
+            .map(|x| x.x509().unwrap().to_owned())
+            .for_each(|cert| {
+                println!("Adding Root certificate {:?}", cert);
+                builder.cert_store_mut().add_cert(cert).unwrap()
+            });
+
+        builder.set_verify(SslVerifyMode::PEER);
+        builder.check_private_key().unwrap();
+        builder.build().into()
+    }
+}
+
+#[derive(Debug)]
+pub struct TlsStream<IO>(SslStream<IO>);
+
+impl<IO> TlsStream<IO>
+where
+    IO: AsyncRead + AsyncWrite,
+{
+    #[cfg(not(feature = "boring-tls"))]
+    pub fn new(ssl: Ssl, stream: IO) -> Self {
+        Self(SslStream::new(ssl, stream).unwrap())
+    }
+}
+
+impl<IO> TlsStream<IO> {
+    pub fn get_alpn_protocol(&self) -> Option<&[u8]> {
+        self.0.ssl().selected_alpn_protocol()
+    }
+
+    pub fn client_identity(&self) -> Option<ClientId> {
+        let cert = self.0.ssl().peer_certificate()?;
+        let peer = cert.subject_alt_names()?.pop()?;
+
+        match ClientId::from_str(peer.dnsname().unwrap()) {
+            Ok(client) => Some(client),
+            Err(_) => None,
+        }
```

Comment:
nit: this `match` could be simplified to
```suggestion
        ClientId::from_str(peer.dnsname().unwrap()).ok()
```